### ISSUE TITLE
Experiments

### ISSUE THREAD

**Issue by tucan9389** [tucan9389/tf2-mobile-pose-estimation#39]
## Table

- [Experiment02](https://github.com/tucan9389/tf2-mobile-pose-estimation/issues/39#issuecomment-641418735)
- [Experiment03](https://github.com/tucan9389/tf2-mobile-pose-estimation/issues/39#issuecomment-644740403)
- [Experiment04](https://github.com/tucan9389/tf2-mobile-pose-estimation/issues/39#issuecomment-645499025)

**Comment by tucan9389** [comment]
# Experiment02

## Dataset Config

- `ai_challenger-gpu.cfg`

## Training Config

- cpm mv2: `experiment02-cpm-gpu.cfg`
- hourglass mv2: `experiment02-hourglass-gpu.cfg`
- simplepose resnet18: `experiment02-sp-resnet18-gpu.cfg`

## Result

### Summary

> Following PCKh@.5s are evaluated with AI Challenger's validation set.

Name | Backbone | PCKh@.5 | tflite size | input shape | output shape
---- | ---- | :----:  | :----:  | :----: | :----: 
CPM | MobileNetV2 | `92.91%` | `4.1MB` | `[1,192,192,3]` | `[1,24,24,14]`
Hourglass | MobileNetV2 | `72.58%` | `1.9MB` | `[1,192,192,3]` | `[1,48,48,14]`
SimplePose | ResNet18 | `97.22%` | `61.5MB` | `[1,192,192,3]` | `[1,48,48,14]`

### Tensorboard

> Preparing...

**Comment by tucan9389** [comment]
# Experiment03

models from https://github.com/edvardHua/PoseEstimationForMobile/tree/master/release

## Result

### Summary

> Following PCKh@.5s are evaluated with AI Challenger's validation set.

Name | Backbone | PCKh@.5 | tflite size | input shape | output shape
---- | ---- | :----:  | :----:  | :----: | :----: 
CPM | MobileNetV2 | `96.29%` | `2.4MB` | `[1,192,192,3]` | `[1,96.96,14]`
Hourglass | MobileNetV2 | `94.93%` | `1.7MB` | `[1,192,192,3]` | `[1,48,48,14]`

**Comment by aitikgupta** [comment]
The values of PCKh@0.5 and tflite size seem to be empty for Experiment03. Are you planning to update it?

**Comment by tucan9389** [comment]
@aitikgupta 

> The values of PCKh@0.5 and tflite size seem to be empty for Experiment03. Are you planning to update it?

Sure. I update it just now.

**Comment by aitikgupta** [comment]
For Experiment02, where could I find the configuration and tflite files?

**Comment by tucan9389** [comment]
> For Experiment02, where could I find the configuration and tflite files?

Here are the tflite files. cfg files are placed on `config/training` folder.

https://github.com/tucan9389/tf2-mobile-pose-estimation/releases/tag/experiment02

**Comment by aitikgupta** [comment]
Thanks, could you also release the corresponding frozen .pb files too?
I wanted to compare them with my trained model.

**Comment by tucan9389** [comment]
> Thanks, could you also release the corresponding frozen .pb files too?
> I wanted to compare them with my trained model.

I just have saved_model now. If you open new issue for this, I’ll share soon. 
Thank you.

**Comment by tucan9389** [comment]
# Experiment04

- various stage number of CPM models
- input shape: `[1,192,192,3]`
- output shape: `[1,24,24,14]`

## Hyperparameters

- number_of_epoch: `200` (about 140.2k step)
→ need to experiment with bigger epoch in the production model

## Results

Stage num | PCKh@.5 | tflite size 
:----: | :----:  | :----: 
1 | `60.16%` | `2.8MB`
2 | `60.24%` | `3.3MB`
3 | `62.38%` | `3.7MB`
4 | `93.27%` | `4.2MB`
5 | `92.82%` | `4.7MB`

# Experiment05

- simple pose with various MobileNetV2

\# | Channels | PCKh@.5 | tflite size
--- | :----: | :----:  | :----: 
1 | `[256, 256]` | `77.58%` | `9.3MB`

## Comments

The first PCKh@.5 doesn't satisfy. I'll reconstruct and experiment again.
